Client/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp

```diff
@@ -6972,16 +6972,22 @@ bool CStaticFunctionDefinitions::GetKeyState(const char* szKey, bool& bState)
 
     CKeyBindsInterface* pKeyBinds = g_pCore->GetKeyBinds();
     const SBindableKey* pKey = pKeyBinds->GetBindableFromKey(szKey);
+    unsigned long keyCode = 0;
+
     if (pKey)
+        keyCode = pKey->ulCode;
+    else if (strcmp(szKey, "capslock") == 0)
+        keyCode = VK_CAPITAL;
+    else if (strcmp(szKey, "numlock") == 0)
+        keyCode = VK_NUMLOCK;
+    
+    if (g_pCore->IsFocused() && keyCode > 0)
     {
-        if (g_pCore->IsFocused())
-        {
-            bState = (::GetKeyState(pKey->ulCode) & 0x8000) ? true : false;
-
-            return true;
-        }
+        DWORD dwFlags = (keyCode == VK_CAPITAL || keyCode == VK_NUMLOCK) ? 0x0001 : 0x8000;

```

Comment:
```suggestion
        DWORD dwFlags = pKey ? 0x8000 : 0x1;
```

Reply:
Your change wouldnt work since pKey is always set for `capslock` (It's added to bindable keys)

Reply:
Wait — does `0x1` return the system-wide enable state, and `0x8000` the current keydown state? 

If that is the case, we cannot make this change and have to do something different

Reply:
How would you do it differently ?